### PR TITLE
Add protocol activation support for Frontends

### PR DESCRIPTION
**Pull request by danprice142**
Register the 'moonlight' URI protocol and handle protocol activations. Adds App::OnActivated and URI parsing to accept parameters (host, appId, appName, desktop, resume, launchOnExit) and a truthy flag parser. New ApplicationState fields track pending protocol host/app/resume and a launchOnExit URI. HostSelectorPage gains HandleProtocolHostSelect to resolve and auto-connect saved hosts; AppPage applies pending app/connect requests after apps are fetched. Streaming exit now launches the return URI (launchOnExit) if provided. Ensures cold-start behavior and avoids interrupting an active stream.

## Summary by CodeRabbit

* **New Features**
  * Added support for opening `moonlight:` links to select a host and launch a specific application.
  * Supports host selection by saved name, identifier, hostname, or IP address.
  * Added resume-session requests and optional links that open automatically when streaming ends.
  * Reuses the existing app window and navigates to the appropriate host or application.

* **Bug Fixes**
  * Malformed, unavailable, or invalid activation requests are safely ignored.
  * Activation requests no longer interrupt an active stream.
  * Improved disconnect handling and verification that streaming applications have stopped.

### CODE REVIEW

**Comment by andygrundman** [comment]
Moonlight UWP is intended for use on Xbox only. I don't know if moonlight-qt has URI support but presumably it would be better if a link launched that client instead.

**Comment by danprice142** [comment]
@andygrundman I was intentionally targeting uwp as there are now three different front ends available Xbox Dev Mode

File: Pages/AppPage.xaml.cpp [hunk 1]
```diff
@@ -118,7 +118,44 @@ void AppPage::OnNavigatedTo(Windows::UI::Xaml::Navigation::NavigationEventArgs^
...
+		int requestedAppId = state->pendingProtocolAppId;
+		std::wstring requestedAppName = state->pendingProtocolAppName;
+		bool resumeRequested = state->pendingProtocolResume;
+		state->pendingProtocolAppId = -1;
+		state->pendingProtocolAppName.clear();
+		state->pendingProtocolResume = false;
+
+		// Dispatched at High priority so this runs after UpdateApps() has filled the Apps list
+		Windows::ApplicationModel::Core::CoreApplication::MainView->CoreWindow->Dispatcher->RunAsync(
+			Windows::UI::Core::CoreDispatcherPriority::High, ref new Windows::UI::Core::DispatchedHandler([this, requestedAppId, requestedAppName, resumeRequested]() {
```
[the remark this answers is not in the document]
Comment (danprice142) [reply Pages/AppPage.xaml.cpp:132]: The lambda captures the ref-class instance through this by strong reference. The dispatcher callback therefore already keeps AppPage alive, and capturing an additional local would be redundant. I’ve left this unchanged.

File: Pages/AppPage.xaml.cpp [hunk 2]
```diff
@@ -118,7 +118,43 @@ void AppPage::OnNavigatedTo(Windows::UI::Xaml::Navigation::NavigationEventArgs^
 		}
 	});
 
-	if (host->AutostartID >= 0 && GetApplicationState()->shouldAutoConnect) {
+	auto state = GetApplicationState();
+	bool hasProtocolRequest = state->pendingProtocolAppId >= 0 || !state->pendingProtocolAppName.empty() || state->pendingProtocolResume;
```
[the remark this answers is not in the document]
Comment (danprice142) [reply Pages/AppPage.xaml.cpp:122]: This behaviour is intentional. A host-only protocol activation follows the existing host-selection behavior if the selected host has an autostart application configured, it launches that application; otherwise, it displays the host’s application list. Explicit appId, appName, or resume parameters take precedence over autostart.

**Comment by andygrundman** [comment]
> @andygrundman I was intentionally targeting uwp as there are now three different front ends available Xbox Dev Mode

I'm still confused, how does one open links on Xbox?

**Comment by danprice142** [comment]
> > @andygrundman I was intentionally targeting uwp as there are now three different front ends available Xbox Dev Mode
> 
> I'm still confused, how does one open links on Xbox?

For example, a frontend can send moonlight:?host=192.168.1.100&appName=Desktop using Windows.System.Launcher.LaunchUriAsync to connect to that host and launch its Desktop application. It can also be tested manually by entering the URI into Microsoft Edge’s address bar on Xbox, or into the Windows Run dialog if testing on PC.


To test launchOnExit, enter moonlight:?host=192.168.1.100&appName=Desktop&launchOnExit=microsoft-edge: into Edge’s address bar/Run dialog. When the stream closes, Moonlight launches microsoft-edge: and returns to Microsoft Edge
